code/modules/reagents/reagent_containers.dm

```diff
@@ -290,4 +336,32 @@ var/list/LOGGED_SPLASH_REAGENTS = list(FUEL, THERMITE)
 	else
 		if(reagents.reagent_list.len)
 			for(var/datum/reagent/R in reagents.reagent_list)
-				to_chat(user, "<span class='info'>[R.volume] units of [R.name]</span>")
\ No newline at end of file
+				to_chat(user, "<span class='info'>[R.volume] units of [R.name]</span>")
+
+
+/obj/item/weapon/reagent_containers/proc/imbibe(mob/user) //Drink the liquid within
+	to_chat(user, "<span  class='notice'>You swallow a gulp of \the [src].</span>")
+	playsound(user.loc,'sound/items/drink.ogg', rand(10,50), 1)
+
+	if(isrobot(user))
+		reagents.remove_any(amount_per_transfer_from_this)
+		return 1
+	if(reagents.total_volume)
+		if(can_drink(user))
+			reagents.reaction(user, INGEST)
+			spawn(5)
+				reagents.trans_to(user, amount_per_transfer_from_this)
+
+	return 1
+
+/obj/item/weapon/reagent_containers/proc/can_drink(mob/user)
+	if (ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.species.chem_flags & NO_DRINK)
+			reagents.reaction(get_turf(H), TOUCH)
+			H.visible_message("<span class='warning'>The contents in [src] fall through and splash onto the ground, what a mess!</span>")
+			reagents.remove_any(amount_per_transfer_from_this)
```

Comment:
make it react with the ground below it first, if I chug thermite as a skeleton and it falls to the ground I should be able to light it